src/cat-data-service/src/legacy_service/health.rs

```diff
@@ -1,23 +1,46 @@
-use crate::service::Error;
-use axum::{routing::get, Router};
+use crate::{
+    service::Error,
+    settings::{RetryAfterParams, RETRY_AFTER_HTTP_DATE_DEFAULT},
+    state::State,
+};
+use axum::{
+    body,
+    extract::Query,
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    routing::{get, put},
+    Router,
+};
+use std::sync::Arc;
 
 use super::handle_result;
 
-pub fn health() -> Router {
+pub fn health(state: Arc<State>) -> Router {
     Router::new()
         .route(
             "/health/ready",
-            get(|| async { handle_result(ready_exec().await) }),
+            get({
+                let state = state.clone();
+                move || async { handle_result(ready_exec(state).await) }
+            }),
         )
         .route(
             "/health/live",
             get(|| async { handle_result(live_exec().await) }),
         )
+        .route(
+            "/health/retry-after",
+            put({
+                let state = state.clone();
+                move |params| async { retry_after_exec(params, state).await }
+            }),
+        )
 }
 
-async fn ready_exec() -> Result<bool, Error> {
+async fn ready_exec(state: Arc<State>) -> Result<bool, Error> {
     tracing::debug!("health ready exec");
 
+    state.event_db.schema_version_check().await?;
```

Comment:
Same question as above

Reply:
In this case, the `/health/ready` endpoint should check that the connection to the database is working as expected. The simplest way that I could think of to implement this with existing functionality is to verify the schema version.